mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoCRUDController.java

```diff
@@ -1122,47 +1121,47 @@ private Error analyzeException(Exception e, final String otherwise, final String
     public CRUDHealth checkHealth() {
         boolean isHealthy = true;
         Collection<MongoConfiguration> configs = dbResolver.getConfigurations();
-        List<String> details = new ArrayList<>(configs.size());
+        Map<String, Object> details = new LinkedHashMap<>();
         DBObject ping = new BasicDBObject("ping", 1);
-
         DB db = null;
         for (MongoConfiguration config : configs) {
             try {
                 db = dbResolver.get(new MongoDataStore(config.getDatabase(), null, null));
                 CommandResult result = db.command(ping);
                 if (!result.get("ok").equals(1.0)) {
                     isHealthy = false;
-                    details.add(new StringBuilder(getMongoConfigDetails(config)).append("=>").append("ping:NOT_OK")
-                            .toString());
                 } else {
-                    details.add(
-                            new StringBuilder(getMongoConfigDetails(config)).append("=>").append("ping:OK").toString());
+                    isHealthy = true;
                 }
             } catch (Exception e) {
                 isHealthy = false;
-                details.add(new StringBuilder(getMongoConfigDetails(config)).append("=>").append("ping_error:")
-                        .append(e.getMessage()).toString());
+                details.put("stackTrace", e);
```

Comment:
Do you know if jackson will serialize the Exception instance nicely?

Reply:
It comes out as follows:

`"stackTrace": "com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting for a server that matches ReadPreferenceServerSelector{readPreference=primary}. Client view of cluster state is {type=STANDALONE, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused}}]",`

So it looks like the full stack trace is not being included, just the message and caused by.  I think that is probably sufficient here, but let me know if you disagree..

Reply:
That looks good to me, though I'd probably change the field name from "stackTrace" to "exception" or something like that.